client/my-sites/stats/stats-purchase/stats-purchase-checkout-redirect.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+	PRODUCT_JETPACK_STATS_MONTHLY,
+	PRODUCT_JETPACK_STATS_PWYW_YEARLY,
+	PRODUCT_JETPACK_STATS_FREE,
+} from '@automattic/calypso-products';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+
+const getStatsPurchaseURL = ( siteSlug: string, product: string, redirectUrl?: string ) => {
+	const checkoutUrl = new URL( 'https://wordpress.com/checkout/' );
+	const checkoutProductUrl = new URL( `${ checkoutUrl }${ siteSlug }/${ product }` );
+
+	// Add redirect_to parameter
+	if ( redirectUrl ) {
+		checkoutProductUrl.searchParams.set( 'redirect_to', redirectUrl ); // TODO: add a redirect URL with query parameter showing proper success/fail banner
+	} else {
+		checkoutProductUrl.searchParams.set(
+			'redirect_to',
+			`https://wordpress.com/stats/${ siteSlug }`
```

Comment:
```suggestion
			`https://wordpress.com/stats/day/${ siteSlug }?statsPurchaseSuccess=1`
```

Reply:
NOTE: we also need to support odyssey return URL. Not a blocker.

Reply:
Captured this todo in [this issue](https://github.com/Automattic/wp-calypso/issues/79329)'s tasklist.